src/roman_datamodels/_typing/maker.py

```diff
@@ -0,0 +1,370 @@
+from __future__ import annotations
+
+import importlib.util as util
+import inspect
+import string
+import sys
+import warnings
+from pathlib import Path
+from typing import get_args, get_type_hints
+
+# This try/except is to recover from pytest screwing with paths when running,
+# meaning the symbolic link from `stnode.pyi` to `_stnode.py` is lost. This
+# block is an attempt to recover from that, or any other similar issues.
+#
+# The reason why `_stnode.py` exists is that Python will not load at runtime
+# any `.pyi` files, even though `.pyi` files have to be valid python. So we
+# trick Python into playing nice. The symbolic link is there so we have one
+# version of they type stubs file, not two.
+try:
+    import _stnode as stubs
+except ModuleNotFoundError:
+    name = "roman_datamodels._typing._stnode"
+    path = Path(__file__).parent / "_stnode.py"
+    spec = util.spec_from_file_location(name, path)
+    stubs = util.module_from_spec(spec)
+    if name not in sys.modules:
+        sys.modules[name] = stubs
+    else:
+        raise ImportError(f"Could not find module {name}")
+    spec.loader.exec_module(stubs)
+
+import astropy.units as u
+import numpy as np
+from astropy.modeling.models import Shift
+from astropy.time import Time
+
+from roman_datamodels import stnode
+from roman_datamodels._typing import enums
+
+NODE_CLASSES_BY_NAME = {cls.__name__: cls for cls in stnode.NODE_CLASSES}
+STUB_CLASSES_BY_NAME = {name: cls for name, cls in inspect.getmembers(stubs, inspect.isclass)}
+ENUM_CLASSES_BY_NAME = {name: cls for name, cls in inspect.getmembers(enums, inspect.isclass)}
+SCALAR_CLASSES_BY_NAME = {cls.__name__: cls for cls in stnode.NODE_CLASSES if issubclass(cls, stnode.TaggedScalarNode)}
+LIST_CLASSES_BY_NAME = {cls.__name__: cls for cls in stnode.NODE_CLASSES if issubclass(cls, stnode.TaggedListNode)}
+
+
+RNG = np.random.default_rng()
+LOW = -1e3
+HIGH = 1e3
+STRING_LENGTH = 32
+TIME_0 = 1577836800.0  # unix date: 2020-01-01
+TIME_1 = 1893456000.0  # unix date: 2030-01-01
+LIST_0 = 1
+LIST_1 = 10
+
+RANDOM_DEFAULT = True
+NONUM = -999999
+NOSTR = "dummy value"
+
+_HEX_ALPHABET = np.array(list(string.hexdigits))
+
+
+DEFAULT_SHAPES = {
+    1: (4096,),
+    2: (4096, 4096),
+    3: (2, 4096, 4096),
+    4: (2, 4, 4096, 4096),
+    5: (2, 8, 16, 32, 32),
+}
+
+REPLACE_NAME = {
+    "pass_": "pass",
+}
```

Comment:
This exists because I cannot use `pass` as a variable name in the way I define the type stub. It is a reserved keyword for Python and bad things happen if I do `pass: foo` as a variable declaration. So this fixes that.